### ISSUE TITLE
The IN infix operator does not work with named query parameters

### ISSUE THREAD

**Issue by dball** [jkk/honeysql#73]
This form doesn't work as expected:

```
(def hsql {:select [:*] :from [:customers] :where [:in :id :?ids]})
(honeysql.core/format hsql {:ids [1 2 3]}) ; => ["SELECT * FROM customers WHERE (id in ?) " [1 2 3]]
```

If this were to work, it would have to result in something of the form:

```
["SELECT * FROM customers WHERE (id IN (?,?,?))" 1 2 3]]
```

If there's any interest in making this work, I'd be happy to take a stab at it. It's unclear offhand if the current behavior is desirable for some reason though, so I thought I'd ask first.

**Comment by MichaelBlume** [comment]
Thanks for the report!

Yeah, I'd be interested in seeing this work, please feel free to take a stab at it =)

**Comment by ddellacosta** [comment]
I think some of this simply has to do with the fact that there are so many ways to do it--for example, here's a workaround for your example:

```clojure
;; using: (def hsql {:select [:*] :from [:customers] :where [:in :id :?ids]})

user=> (honeysql.core/format (assoc-in hsql [:where 2] [1, 2, 3]))
["SELECT * FROM customers WHERE (id in (1, 2, 3))"]
user=> (honeysql.core/format (assoc-in hsql [:where 2] ["one", "two", "three"]))
["SELECT * FROM customers WHERE (id in (?, ?, ?))" "one" "two" "three"]
```

But yeah, seems like something that worked correctly using your example code would be nice.

**Comment by dball** [comment]
Yeah, I just use inline values for the most part without issue, but I am in the process of experimenting with extracting some hsql queries into vars and the param version would be the most readable form. I'll see if I can come up with something.

**Comment by MichaelBlume** [comment]
Fixed by #74
